### PR TITLE
Fix FreeBSD HardwareMonitor compilation under FreeBSD

### PR DESCRIPTION
**Pull request by xbeaudouin**
Two fixes for FreeBSD HardwareMonitor compilation under FreeBSD... (and one to say to CLang to stop complaining...)

### CODE REVIEW

File: hardware/HardwareMonitor.h [hunk 1]
```diff
@@ -96,6 +96,9 @@ class CHardwareMonitor : public CDomoticzHardwareBase
 	float GetProcessMemUsage();
 	float GetMemUsageLinux();
 #endif
+#if defined (__FreeBSD__)
+	float GetMemUsageLinux();
+#endif
 #if defined (__FreeBSD__) || defined(__OpenBSD__)
```
Comment (kiddigital) [inline hardware/HardwareMonitor.h:102]: @xbeaudouin , how does this relate to this OR ? We compile 2 memory function for FreeBSD (GetMemUsageLinux and GetMemUsgaeOpenBSD)?

**Review by kiddigital — COMMENTED** [review]
Good catch... maybe here explicit set the sOStype variable to Unknown (again and/or make the variable at the top empty to start with) to make things even more readable.

File: hardware/HardwareMonitor.h (hunk 1 above)
Comment (xbeaudouin) [reply hardware/HardwareMonitor.h:102]: Well the original compilation looks for GetMemUsageLinux() as you see in the following error while compiling.
```
hardware/HardwareMonitor.cpp:743:26: error: out-of-line definition of 'GetMemUsageLinux' does not match any declaration in 'CHardwareMonitor'
        float CHardwareMonitor::GetMemUsageLinux()
                                ^~~~~~~~~~~~~~~~
1 warning and 1 error generated.
```

Aslo on this code there is also specific FreeBSD stuff. So FreeBSD seems to need this function.

**Comment by xbeaudouin** [comment]
Well Clang 8.x is very picky about switch().
It needs a default.
Maybe we should do

```
case default : 
  sOStype= "Unknown';
  break; 
```

What do you think ?

**Comment by kiddigital** [comment]
> Well Clang 8.x is very picky about switch().
> 
> It needs a default.
> 
> Maybe we should do
> 
> 
> 
> ```
> 
> case default : 
> 
>   sOStype= "Unknown';
> 
>   break; 
> 
> ```
> 
> 
> 
> What do you think ?
> 
> 

Sounds good 👍

**Comment by xbeaudouin** [comment]
This is done :D

File: hardware/HardwareMonitor.h (hunk 1 above)
Comment (kiddigital) [reply hardware/HardwareMonitor.h:102]: @xbeaudouin ok clear. Maybe than it is more clear to make it something like
`#if defined (__linux__) || defined (__FreeBSD__)`
and remove the function declaration 'float GetMemUsageLinux()' from the lines above (97)

This way it is only referenced once in the header file and it is visible at once for which systems it is needed.

**Comment by kiddigital** [comment]
BTW, thx for testing and helping out to make sure my previous changes from #4393 work on FreeBSD :)

**Comment by gizmocuz** [comment]
Thank you both !!

**Pull request merged**